smack-extensions/src/main/java/org/jivesoftware/smackx/avatar/element/DataExtension.java

```diff
@@ -0,0 +1,84 @@
+/**
+ *
+ * Copyright 2017 Fernando Ramirez
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.avatar.element;
+
+import org.jivesoftware.smack.packet.ExtensionElement;
+import org.jivesoftware.smack.util.XmlStringBuilder;
+import org.jivesoftware.smack.util.stringencoder.Base64;
+import org.jivesoftware.smackx.avatar.UserAvatarManager;
+
+/**
+ * Data extension element class.
+ * 
+ * @author Fernando Ramirez
+ * @see <a href="http://xmpp.org/extensions/xep-0084.html">XEP-0084: User
+ *      Avatar</a>
+ */
+public class DataExtension implements ExtensionElement {
+
+    public static final String ELEMENT = "data";
+    public static final String NAMESPACE = UserAvatarManager.DATA_NAMESPACE;
+
+    private final byte[] data;
+
+    /**
+     * Data Extension constructor.
+     * 
+     * @param data
+     */
+    public DataExtension(byte[] data) {
+        this.data = data;
+    }
+
+    /**
+     * Get data.
+     * 
+     * @return a copy of the immutable data
+     */
+    public byte[] getData() {
+        return data.clone();
+    }
+
+    /**
+     * Get data as String.
+     * 
+     * @return the data as String
+     */
+    public String getDataAsString() {
```

Comment:
This should be called `getDataBase64Encoded()`. To make it clear that the string representation of the data is base64 encoded.